webrtc/RTCConfiguration-iceServers.html

```diff
@@ -0,0 +1,570 @@
+        - or if parsing based on the syntax defined in [ RFC7064] and [RFC7065] fails,
+          throw a SyntaxError.
+
+    [RFC7064] URI Scheme for the Session Traversal Utilities for NAT (STUN) Protocol
+    3.1.  URI Scheme Syntax
+      stunURI       = scheme ":" host [ ":" port ]
+      scheme        = "stun" / "stuns"
+
+    [RFC7065] Traversal Using Relays around NAT (TURN) Uniform Resource Identifiers
+    3.1.  URI Scheme Syntax
+      turnURI       = scheme ":" host [ ":" port ]
+                      [ "?transport=" transport ]
+      scheme        = "turn" / "turns"
+      transport     = "udp" / "tcp" / transport-ext
+      transport-ext = 1*unreserved
+   */
+  config_test(makePc => {
+    assert_throws('SyntaxError', () =>
+      makePc({ iceServers: [{
+        urls: 'relative-url'
+      }] }));
+  }, 'with relative url should throw SyntaxError');
+
+  config_test(makePc => {
+    assert_throws('SyntaxError', () =>
+      makePc({ iceServers: [{
+        urls: 'http://example.com'
+      }] }));
+  }, 'with http url should throw SyntaxError');
+
+  config_test(makePc => {
+    assert_throws('SyntaxError', () =>
+      makePc({ iceServers: [{
+        urls: 'turn://example.org/foo?x=y'
+      }] }));
+  }, 'with invalid turn url should throw SyntaxError');
+
+  config_test(makePc => {
+    assert_throws('SyntaxError', () =>
+      makePc({ iceServers: [{
+        urls: 'stun://example.org/foo?x=y'
+      }] }));
+  }, 'with invalid stun url should throw SyntaxError');
+
+  config_test(makePc => {
+    const pc = makePc({ iceServers: [{
+      urls: [],
+      credentialType: 'password'
+    }] });
+
+    const { iceServers } = pc.getConfiguration();
+    assert_equals(iceServers.length, 1);
+
+    const server = iceServers[0];
+    assert_array_equals(server.urls, []);
+    assert_equals(server.credentialType, 'password');
+  }, `with empty urls and credentialType password should succeed`);
+
+  config_test(makePc => {
+    const pc = makePc({ iceServers: [{
+      urls: [],
+      credentialType: 'oauth'
+    }] });
+
+    const { iceServers } = pc.getConfiguration();
+    assert_equals(iceServers.length, 1);
+
+    const server = iceServers[0];
+    assert_array_equals(server.urls, []);
+    assert_equals(server.credentialType, 'oauth');
+  }, `with empty urls and credentialType oauth should succeed`);
+
+  config_test(makePc => {
+    assert_throws(new TypeError(), () =>
+      makePc({ iceServers: [{
+        urls: [],
+        credentialType: 'invalid'
+      }] }));
+  }, 'with invalid credentialType should throw TypeError');
+
+  // token credentialType was removed from the spec since 20170508
+  config_test(makePc => {
+    assert_throws(new TypeError(), () =>
+      makePc({ iceServers: [{
+        urls: [],
+        credentialType: 'token'
+      }] }));
+  }, 'with credentialType token should throw TypeError');
+
+  // Blink and Gecko fall back to url, but it's not in the spec.
```

Comment:
I added this comment in 089c33e7267854a391cfd8ce38929c52425f1252 to explain the `'{ iceServers: [{ url: "stun:stun1.example.net" }] }': new TypeError,` line that followed but which isn't in this file.

It's an historical test, to make sure that some test is failing as long as a formerly-standard thing hasn't been removed from implementations. These are often in historical.html tests. I'm adding one for WebRTC in https://github.com/w3c/web-platform-tests/pull/6686, once that's landed the test for non-support of `url` could be moved there.